pkg/release/publish.go

```diff
@@ -294,6 +295,26 @@ func (p *Publisher) VerifyLatestUpdate(
 	return true, nil
 }
 
+func IsUpToDate(oldVersion, newVersion semver.Version) bool {
+	oldPre := oldVersion.Pre
+	newPre := newVersion.Pre
+
+	// Verfy specific use case in our tagging logic:
+	// 1.30.0-rc.2.10+00000000000000
+	// needs to be considered lower than
+	// 1.30.0-11+00000000000000
+	// which is not given by newVersion.LTE(oldVersion) below.
+	if len(oldPre) == 3 && // [rc 2 10]
+		oldPre[0].String() == "rc" &&
+		len(newPre) == 1 && // [11]
+		newPre[0].IsNum &&
+		newPre[0].Compare(oldPre[2]) > 0 {
```

Comment:
This is not going to work because once `.0` is out, the counter will reset, so we will have something like `1.30.0-1+00000000000000`. Let's say we have this scenario:

- `1.30.0-1+00000000000000` - we'll have `[1]`
- `1.30.0-rc.2.10+00000000000000` - we'll have `[rc 2 10]`

This is going to fail this line because `1.Compare(10)` will return value `< 0`, but the version is actually newer.

Here's a concrete example of this: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-build-1-30/1781291580624211968

```
level=info msg="Not updating version, because v1.30.0-4+e7178656ad89f7 <= v1.30.0-rc.2.11+a2b4393c758dc3"
level=info msg="Skipping k8s-beta.txt for v1.30.0-4+e7178656ad89f7 because it does not need to be updated"
```

I did some napkin maths at this time, I'm not sure if we can just drop this line, but I can double check if needed.

Reply:
@saschagrunert I think there's one more edge case that can lead to a problem very similar to https://github.com/kubernetes/sig-release/issues/2333. If you have something like this:

- `oldVersion`: `1.29.0-rc.0.20+00000000000000`
- `newVersion`: `1.28.1-2+00000000000000`

If I'm not mistaken, this is going to result into `false`, getting the marker updated even though 1.29.0-rc.0 is newer than 1.28.1.

What I don't know is if we can realistically hit this case. I'm thinking this might be a problem after cutting rc.0. Then building any older version might end up overwriting the `latest.txt` file with that older version.

Reply:
@xmudrii I think that's somewhat artificial because `newVersion` would be on a different release branch than `oldVersion`. I don't know, we may end up with that scenario so I added another check + test to fix that.

Reply:
Yeah, I also think it's artificial now that I research a bit more, but I guess it's likely safer to cover it as well.